app/assets/javascripts/pretty-text/addon/sanitizer.js

```diff
@@ -146,7 +150,8 @@ export function sanitize(text, allowLister) {
         }
 
         if (tag === "iframe" && name === "src") {
-          return "-STRIP-";
+          // This iframe is not allowed
+          return "";
```

Comment:
Instead of replacing the `src` attribute with `-STRIP-`, just skip it altogether.

The `IFRAME_REGEXP` will (hopefully) catch all `iframe` tags without a `src` attribute.